### PR TITLE
Don't create validations for calibration observations

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

**Comment by toddburnside** [comment]
@tpolecat If you are using the validations in the observation status process, you might want to review this and make sure it doesn't affect you.

File: modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala [hunk 1]
```diff
@@ -713,21 +721,24 @@ object ObservationService {
...
 
-        val initialMap = 
-          for {
-            (genVals, op) <- generatorValidations
-            cfpVals       <- cfpValidations
-            itcVals       <- op.filter(_ => cfpVals.isEmpty).foldMapM(itcValidations) // only compute this if cfp and gen are ok
-            bandVals      <- validateScienceBand
-          } yield (genVals |+| itcVals |+| cfpVals |+| bandVals)
+        obsInfo.flatMap: info =>
+          if (info.role.isDefined) List.empty.pure // it's a calibration
```
Comment (cquiroz) [inline modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:725]: I still wonder about no warnings. Calibrations can be ill defined, e.g. no guide star

File: modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala [hunk 2]
```diff
@@ -1484,6 +1484,12 @@ trait DatabaseOperations { this: OdbSuite =>
       .use(_.prepareR(command).use(_.execute(role, tid).void))
   }
 
+  def setObservationCalibratioRole(oid: Observation.Id, role: CalibrationRole): IO[Unit] = {
```
Comment (cquiroz) [inline modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala:1487]: I think this was defined in `CalibrationService` already

Comment (toddburnside) [reply modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala:1487]: Ah. Can I call a service method from the tests?

File: modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala (hunk 1 above)
Comment (toddburnside) [reply modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:725]: Hmm. Even if they are automatically generated? And if so, what can the user do about that? Who should be notified and how?

**Comment by tpolecat** [comment]
In terms of workflow it seems like calibrations shouldn't really count, so this makes sense to me.

File: modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala (hunk 2 above)
Comment (cquiroz) [reply modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala:1487]: Sure, though I think it ought to be a static method

File: modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala (hunk 1 above)
Comment (cquiroz) [reply modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:725]: Maybe a question for user?

Comment (toddburnside) [reply modules/service/src/main/scala/lucuma/odb/service/ObservationService.scala:725]: Andy commented on the shortcut ticket:

> That's a good question. I am optimistically hopeful that all of our automatically generated calibration observations will be perfect. I could imagine that if all targets were accidentally deleted from the Specphot standard storage program, or a target was added without an SED or brightness information those could cause trouble, but we probably don't want to bother users about that.

> I don't know if we need to solve this now, but we may need a way to give a warning to ourselves if an automatic calibration fails to generate

**Review by cquiroz — APPROVED** [review]
LGTM

**Pull request merged**